src/components/Markings/components/DragableSquare.js

```diff
@@ -0,0 +1,241 @@
+import React from 'react'
+import {
+    PanResponder,
+} from 'react-native'
+import PropTypes from 'prop-types'
+import { 
+    G,
+    Circle,
+    Rect
+} from 'react-native-svg'
+import CloseButtonSVG from './CloseButtonSVG'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import * as drawingScreenAction from '../../../actions/drawingScreen'
+
+const mapStateToProps = (state, ownProps) => {
+    const { x, y, width, height, color } = state.drawingScreen.shapes[ownProps.index]
+    return {
+        x,
+        y,
+        width,
+        height,
+        color
+    }
+}
+
+const mapDispatchToProps = (dispatch) => ({
+    drawingScreenActions: bindActionCreators(drawingScreenAction, dispatch)
+})
+
+
+class DragableSquare extends React.Component {
+
+    createPanResponder = (onPanResponderMove, onPanResponderRelease) => PanResponder.create({
+        // Ask to be the responder:
+        onStartShouldSetPanResponder: () => this.props.isEditable,
+        onStartShouldSetPanResponderCapture: () => this.props.isEditable,
+        onMoveShouldSetPanResponder: () => this.props.isEditable,
+        onMoveShouldSetPanResponderCapture: () => this.props.isEditable,
+        onPanResponderTerminationRequest: () => this.props.isEditable,
+        onShouldBlockNativeResponder: () => this.props.isEditable,
+        onPanResponderMove,
+        onPanResponderRelease,
+    });
+
+    constructor(props) {
+        super(props)
+
+        /**
+         * Even though we receive props for positioning,
+         * we use state to animate pan gestures. We do this 
+         * so we don't update global state on pan gestures
+         */
+        this.state = { 
+            x: props.x,
+            y: props.y,
+            width: props.width,
+            height: props.height
+        }
+        
+        /**
+         * When this component is dragable, each corner of the square gets a circle.
+         * Each Circle receives a pan responder that mutates the square,
+         */
+        this.topLeftResponder = this.createPanResponder(
+            (evt, gestureState) => {
+                const { dx, dy } = gestureState;
+                this.setState({
+                    x: this.props.x + dx,
+                    y: this.props.y + dy,
+                    width: this.props.width - dx,
+                    height: this.props.height - dy
+                })
+            },
+            (evt, gestureState) => {
+                const { dx, dy } = gestureState;
+                this.props.drawingScreenActions.mutateShapeAtIndex({
```

Comment:
This feels a bit odd that the props are mutated here, as the state is updated above, I'm assuming on the `onPanResponderMove`. What is happening here with `mutateShapeAtIndex`?

Reply:
So this is sort of what I'm trying to explain in the comments on line 50. 

Unfortunately, because react-native doesn't support animating SVGs, you need to either do animations by updating the state or props of components with the pan responder callbacks or by modifying components props directly using `setNativeProps`. Both are fairly hacky and each have their pros and cons.

Updating props and state: Is fairly verbose and the way you are "supposed to" update components, however it's performance can suffer when a lot of updates are happening.

Using `setNativeProps`: More performant, much harder to keep track of state. I had played around with this and was having a lot of trouble getting it to work well.

So I elected to use updating props and state, but I still wanted to cut down on the number of re-renders I do. So I used a component localized state for when he user is dragging the component to mutate it and once the gesture ends then we update global state (calling `mutateShapeAtIndex`) which updates all the other components that care about it (`DrawingScreen`).

To sum up, I use `setState` for drag animations and once the animation is complete , I update redux state on how the shape was mutated. I do all this to save on performance

Reply:
I think that's a good solution. If the code hasn't changed, PFE calls `setState` whenever the shape changes at all, whether on drag or resize, and I always thought there might be a better solution to this. I think it is a good goal to cut down on the number of rerenders taking place.